src/metabase/models/model_index.clj

```diff
@@ -0,0 +1,134 @@
+(ns metabase.models.model-index
+  (:require
+   [clojure.string :as str]
+   [metabase.db.connection :as mdb.connection]
+   [metabase.models.card :refer [Card]]
+   [metabase.models.interface :as mi]
+   [metabase.query-processor :as qp]
+   [metabase.sync.schedules :as sync.schedules]
+   [metabase.util.cron :as u.cron]
+   [metabase.util.i18n :refer [trs]]
+   [metabase.util.log :as log]
+   [toucan.models :as models]
+   [toucan2.core :as t2]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;; model lifecycle ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(models/defmodel ModelIndex :model_index)
+(models/defmodel ModelIndexValue :model_index_value)
```

Comment:
We should use the toucan2 way to define a model: 

https://github.com/metabase/metabase/blob/8f5f36720ab2ee2a70bfed9d150536b6aca9cda8/src/metabase/models/card.clj#L34-L39


```
(methodical/defmethod t2/table-name :model/ModelIndex [_model] :model_index)
(methodical/defmethod t2/table-name :model/ModelIndexValue [_model] :model_index_value)

(derive :model/ModelIndex :metabase/model)
(derive :model/ModelIndexValue :metabase/model)
```

Reply:
a bonus point to start using `:model/ModelIndex` instead of `Model` too.